### ISSUE TITLE
elfutils no longer cross-compiles (target: aarch64-linux, buildHost: x86_64-linux)

### ISSUE THREAD

**Issue by colemickens** [NixOS/nixpkgs#167831]
### Describe the bug

```
# snip ...
checking lex library... none needed
checking whether yytext is a pointer... no
checking for aarch64-unknown-linux-gnu-ar... aarch64-unknown-linux-gnu-ar
checking the archiver (aarch64-unknown-linux-gnu-ar) interface... ar
checking for aarch64-unknown-linux-gnu-readelf... aarch64-unknown-linux-gnu-readelf
checking for aarch64-unknown-linux-gnu-nm... aarch64-unknown-linux-gnu-nm
checking whether gcc supports __attribute__((visibility()))... yes
checking whether gcc supports __attribute__((gcc_struct))... no
checking whether gcc supports -fPIC... yes
checking whether gcc supports -fPIE... yes
checking whether gcc supports -Wl,-z,defs... yes
checking whether the compiler generates build-ids... no
configure: WARNING: compiler doesn't generate build-id by default
checking whether gcc supports -Wl,-z,relro... yes
checking for __thread support... yes
checking whether gcc provides stdatomic.h... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking whether fts.h is bad when included (with LFS)... no
checking whether to add -D_FORTIFY_SOURCE=2 to CFLAGS... yes
checking for library containing gzdirect... no
configure: error: zlib not found but is required

```

This is despite the fact that `zlib` is listed as a `buildInput`:

https://github.com/NixOS/nixpkgs/blob/b63eab19daa7ec26fc1b040f610b4af06d84b232/pkgs/development/tools/misc/elfutils/default.nix#L49-L61

This is unfortunate because it effectively blocks being able to cross-compile mobile-nixos boot images.

Unfortunately, I don't have a lot of experience with cross-compiling to know... where to really look? or what to try. Any tips appreciated! Cheers

**Comment by colemickens** [comment]
I'm kind of confused, I'm not seeing any changes in nixpkgs that seem related so far?

**Comment by bobby285271** [comment]
Probably related to #167395?

**Comment by Mindavi** [comment]
Yeah, probably fixed in that pr 🙂. Issue is that it wasn't producing shared libraries without that pr, so the zlib derivation was kinda empty

**Comment by Mindavi** [comment]
Seems to be fixed in master (tested on e1212c4c2812e92d1c7cbed63eb0e417de89a430)
